src/main/java/hudson/remoting/Checksum.java

```diff
@@ -54,4 +60,38 @@ public int hashCode() {
     public String toString() {
         return String.format("%016X%016X",sum1,sum2);
     }
+
+    /**
+     * Returns the checksum for the given file.
+     */
+    static Checksum forFile(File file) throws IOException {
+        return forURL(file.toURI().toURL());
+    }
+
+    /**
+     * Returns the checksum for the given URL.
+     */
+    static Checksum forURL(URL url) throws IOException {
+        try {
+            MessageDigest md = MessageDigest.getInstance(JarLoaderImpl.DIGEST_ALGORITHM);
+            Util.copy(url.openStream(), new DigestOutputStream(new NullOutputStream(), md));
+            return new Checksum(md.digest(), md.getDigestLength() / 8);
+        } catch (NoSuchAlgorithmException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    private static class NullOutputStream extends OutputStream {
```

Comment:
Trying to keep remoting small in terms of dependency footprint as it's used by CLI and etc.